internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -307,6 +342,22 @@ func (r *delegatingResolver) updateTargetResolverState(state resolver.State) err
 		logger.Infof("Addresses received from target resolver: %v", state.Addresses)
 	}
 	r.targetResolverState = &state
+	// If no addresses returned by resolver have network type as tcp , do not
+	// wait for proxy update.
+	if !tcpAddressPresent(r.targetResolverState) {
+		return r.cc.UpdateState(*r.targetResolverState)
+	}
+
+	if len(r.proxyAddrs) == 0 {
+		go func() {
+			r.childMu.Lock()
+			defer r.childMu.Unlock()
+			if _, ok := r.proxyResolver.(nopResolver); ok {
+				r.proxyResolver, _ = r.proxyURIResolver(resolver.BuildOptions{})
```

Comment:
We should not silently ignore this error. The channel may end up waiting forever without receiving any error. This line would also set `r.proxyResolver` to `nil` causing a panic later. 

I think we should inform the channel about the error. The only way I can think of is to call `ClientConn.ReportError`. We may also want to avoid rebuilding the proxy resolver if it has failed once. @easwars @dfawley what do you think?

Reply:
> This line would also set r.proxyResolver to nil causing a panic later.

We should not assign the returned value to `r.proxyResolver` unless the error is nil.